commit f1129012827b8bf33d9e7ac535049d048f726757
Author: Lon Hohberger <lhh@redhat.com>
Date:   Thu Jun 23 10:36:09 2011 -0400

    rgmanager: Fix bad passing of SFL_FAILURE up
    
    We were handling failures at the 'highest' independent subtree
    level instead of lowest; this patch fixes that.
    
    Resolves: rhbz#716231
    
    Signed-off-by: Lon Hohberger <lhh@redhat.com>
    Reviewed-by: Bob Peterson <rpeterso@redhat.com>

diff --git a/rgmanager/src/daemons/restree.c b/rgmanager/src/daemons/restree.c
index 3892ab9..a13f907 100644
--- a/rgmanager/src/daemons/restree.c
+++ b/rgmanager/src/daemons/restree.c
@@ -1548,7 +1548,7 @@ _res_op_internal(resource_node_t __attribute__ ((unused)) **tree,
 		  and all our children as failed and return a flag stating
 		  that this section is recoverable apart from siblings in
 		  the resource tree. */
-		if (op == RS_STATUS && (rv & (SFL_FAILURE|SFL_RECOVERABLE)) &&
+		if (op == RS_STATUS && (rv & SFL_FAILURE) &&
 		    (node->rn_flags & RF_INDEPENDENT)) {
 
 			rte = restart_threshold_exceeded(node->rn_restart_counter);
